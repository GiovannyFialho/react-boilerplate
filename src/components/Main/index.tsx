import { Wrapper, Title, Description, Logo, Illustration } from "./styles";

const Main = ({
    title = "React Boilerplate",
    description = "Typescript, ReactJS, NextJS, Styled Components, Jest e Storybook"
}) => (
    <Wrapper>
        <Logo
            src="/img/logo.svg"
            alt="Imagem de um átomo e React avaçado escrito ao lado."
            title="Logo react avaçado"
        />
        <Title>{title}</Title>
        <Description>{description}</Description>
        <Illustration
            src="/img/hero-illustration.svg"
            alt="Desenvolvedor de frente para a tela com o código"
            title="Ilustração curso de React avançado"
        />
    </Wrapper>
);

export default Main;
