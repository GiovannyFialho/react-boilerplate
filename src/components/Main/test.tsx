import { render, screen } from "@testing-library/react";

import Main from ".";

describe("<Main />", () => {
    it("renderização de título", () => {
        const { container } = render(<Main />);

        expect(
            screen.getByRole("heading", { name: /React Boilerplate/i })
        ).toBeInTheDocument();

        expect(container.firstChild).toMatchSnapshot();
    });

    it("renderização de backgroundColor correto", () => {
        const { container } = render(<Main />);

        expect(container.firstChild).toHaveStyle({
            "background-color": "#06092b"
        });
    });
});
