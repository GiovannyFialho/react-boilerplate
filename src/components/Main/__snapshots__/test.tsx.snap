// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> renderização de título 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 3rem;
}

.c1 {
  width: 25rem;
  margin-bottom: 8rem;
}

.c2 {
  font-size: 4.5rem;
  margin-bottom: 2rem;
}

.c3 {
  font-size: 2rem;
  font-weight: 400;
}

.c4 {
  margin-top: 3rem;
  width: min(30rem,100%);
}

<main
  class="c0"
>
  <img
    alt="Imagem de um átomo e React avaçado escrito ao lado."
    class="c1"
    src="/img/logo.svg"
    title="Logo react avaçado"
  />
  <h1
    class="c2"
  >
    React Boilerplate
  </h1>
  <h2
    class="c3"
  >
    Typescript, ReactJS, NextJS, Styled Components, Jest e Storybook
  </h2>
  <img
    alt="Desenvolvedor de frente para a tela com o código"
    class="c4"
    src="/img/hero-illustration.svg"
    title="Ilustração curso de React avançado"
  />
</main>
`;
